packages/foam-core/src/markdown-provider.ts

```diff
@@ -55,8 +54,8 @@ export function createNoteFromMarkdown(
 
     if (node.type === 'wikiLink') {
       links.push({
-        to: node.value as string,
-        text: node.value as string,
+        type: 'wikilink',
+        slug: node.value as string,
```

Comment:
I'm wondering why there's only slug for each wikiLink? Currently slug is basically equivalent to the file name, meaning wikiLinks that are titles are not going to get identified. I think a small adjustment of this PR (namely `getNotes` (`note-graph.ts:95`)) will enable Foam to allow both [[Title Case]] and [[file-case]] links, as mentioned in this comment: https://github.com/foambubble/foam/pull/103#issuecomment-657257591

Reply:
The way I approached it is to have different type of links for different situations we encounter in the md document. 
For instance, if we find a `[[my-note]]` vs `[[My Note]]` vs `[text](http://link)` link, we actually model them differently (as they are different things: the first one a standard wikilink, the second a title-based-wikilink (for lack of a better name), the last one a regular url link.
At this phase of the parsing we are just modeling the document (atm `Note` and `GraphNote` are basically the same as we don't need extra stuff, that might change in the future).

Of course other ways of doing this are possible (e.g. `wikilink` type could include both with some sort of qualifier), but atm I believe it's better to be more explicit. To be clear, I haven't run this with the rest of the team (except via the PR review), so it's not a "project strategy" more than it is my personal approach.

Related to #103 , another field that might be added in this specific type is `text` or `label` to support wikilinks in the format `[[my-note|this is the label]]`, we removed that from the original implementation because we don't parse it at the moment and it was a bit confusing. Same would apply for an anchor (`[[my-note#my-section]]`).